Update api-management-subscriptions.md (#120943)
Azure/azure-content: articles/api-management/api-management-subscriptions.md

> [!CAUTION]
> Use care when configuring a product or an API that doesn't require a subscription. This configuration may be overly permissive and may make an API more vulnerable to certain [API security threats](mitigate-owasp-api-threats.md#security-misconfiguration).

> [!NOTE]

> > Open products have the **Requires subscription** setting disabled, which means users do not need to subscribe to it. For this reason, open products are not displayed on **Products** page of the developer portal.

Reviewer: ```suggestion
> Open products have the **Requires subscription** setting disabled, which means that users don't need to subscribe to it. For this reason, open products aren't displayed on the **Products** page of the developer portal.
```
Minor edits for Microsoft style
